File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> ans;
        for(int i = 0; i < nums.size(); i++){
            int x = nums[i];
            int comp = target - x;
            if(ans.count(comp)){
                return {ans[comp] , i};
            }
            ans[x] = i;
        }
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        curr = []
        ans = []
        def dfs(i, cnum, curr):
            if cnum == target:
                ans.append(curr.copy())
                return
            if cnum > target:
                return
            if i == len(nums):
                return
            curr.append(nums[i])
            dfs(i,cnum+nums[i], curr)
            curr.pop()
            dfs(i+1,cnum, curr)
        dfs(0,0,[])
        return ans

        